### DISCUSSION TITLE
Organization defaults/templates for repositories (GitHub Enterprise)

### DISCUSSION THREAD

**Discussion by beanssoft** [community/community#13991] [Enterprise]
**Feature Request:** In my team we'd love to be able to configure organization defaults (GitHub Enterprise) for the following, so we don't have to hand configure them for every new repo:

Branch protection rules
Webhooks
Merge button strategy
Auto-delete head branches

Thanks in advance!

**Comment by dgteixeira** [comment]
This is a requirement on our side as well!

**Comment by pfcarrier** [comment]
This might be of interest to people looking for such functionality and was recommended to us by our Github rep.
* https://github.com/github/safe-settings

We ended up implementing it for a 1000+ repos orgs and this work ok.  Only issue is that the "dry-run" and the Cron based run take ~2 hours to run but the events based bits allow this to be a non-issue for now, just be sure that you implement it as two different Github App ( one for the reaction to events such as repos creation, branch protection modification, etc and another distinct one for events such as push and with the CRON env var ).  Hope it help.

**Comment by Snailedlt** [comment]
There's currently no way to set default merge strategy on GitHub in general, and I didn't find any feature requests for it either. So I made a feature request for it here: https://github.com/github-community/community/discussions/20583

Please go vote and comment if you agree with the suggested feature(s)

**Comment by vinod-soni-microsoft** [comment]
+1

**Comment by patrick-knight** [comment]
Managing branch protections at the org is now available in public beta with [repository rules.](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-organization-settings/managing-rulesets-for-repositories-in-your-organization)

**Reply by ataylor-admin** [reply]
When will this be available in prod?

**Reply by patrick-knight** [reply]
[Rules went GA July 24th](https://github.blog/2023-07-24-github-repository-rules-are-now-generally-available/) and will be available later this year in GitHub Enterprise Server.

**Reply by rosshosman** [reply]
This should be at the team level so teams can enforce it across their org.

**Reply by navdotnetreqs** [reply]
Yes but team level earns less money.

**Comment by kolizo12** [comment]
> [Rules went GA July 24th](https://github.blog/2023-07-24-github-repository-rules-are-now-generally-available/) and will be available later this year in GitHub Enterprise Server.

Is this now available on Github enterprise?

**Reply by patrick-knight** [reply]
Rules will be in the next major release of GitHub Enterprise Server.

**Comment by jamonkko** [comment]
@patrick-knight the org level rules are great 👍 

Is there any plans to support org level default for that "Auto-delete head branches" to not have to toggle on the automatic deletion of merged branches for each new org repository separately?

**Reply by patrick-knight** [reply]
Not at the moment. 

But if were to go to explore merge controls like this in rules, would your expectation be to have this behavior be per repo or per branch?

**Reply by jamonkko** [reply]
I think I would expect it to be just default setting for new repos created under organization, so per repo? 
Of course other rules should still override it, e.g. protected branches would not get auto deleted.
